### PR TITLE
Refactor json policy tests

### PR DESCRIPTION
**Pull request by memark**
## Description of changes

Preparatory refactoring work for issue below. Next PR will add new functionality.

Split a test into two, add another one, rename some functions.

## Issue #, if available

https://github.com/cedar-policy/cedar/issues/950

## Checklist for requesting a review

The change in this PR is (choose one, and delete the other options):

- [x] A change "invisible" to users (e.g., documentation, changes to "internal" crates like `cedar-policy-core`, `cedar-validator`, etc.)

I confirm that this PR (choose one, and delete the other options):

- [x] Does not update the CHANGELOG because my change does not significantly impact released code.

I confirm that [`cedar-spec`](https://github.com/cedar-policy/cedar-spec) (choose one, and delete the other options):

- [x] Does not require updates because my change does not impact the Cedar formal model or DRT infrastructure.

### CODE REVIEW

File: cedar-policy-cli/src/lib.rs [hunk 1]
```diff
@@ -1189,7 +1193,7 @@ fn read_json_policy(filename: Option<impl AsRef<Path> + std::marker::Copy>) -> R
         Report::new(err).with_source_code(NamedSource::new(name, json_source.clone()))
     };
 
-    match Policy::from_json(None, json.clone()).map_err(err_to_report) {
```
Comment (memark) [inline cedar-policy-cli/src/lib.rs:1192]: This had no effect, since it was matched with `Err(_)` further down.

**Review by khieta — APPROVED** [review]
Thanks for the additional test cases!

Just a minor comment: I think it'd be fine to have just one `json-format` sandbox like before. Or you could call the new sandboxes `json-authorize` and `json-check-parse` - I think that `authorize` and `check_parse` may be too general. (Although, to be fair, our current sandbox naming is all over the place.)

**Review by shaobo-he-aws — APPROVED** [review]

**Review by andrewmwells-amazon — APPROVED** [review]
LGTM

**Comment by memark** [comment]
> Just a minor comment: I think it'd be fine to have just one `json-format` sandbox like before. Or you could call the new sandboxes `json-authorize` and `json-check-parse` - I think that `authorize` and `check_parse` may be too general. (Although, to be fair, our current sandbox naming is all over the place.)

@khieta, in my view, the ability to read json policies is kind of a cross-cutting concern, not specific to any one subcommand. Hence I would prefer to test it using only `check-parse`, which is closer to the metal than `authorize`, but I felt it a bit out of place to remove the `authorize` test. What are you thoughts here?

Out of your alternatives, I'd prefer the two separate json-prefixed sandboxes.

**Review by khieta — APPROVED** [review]
New names look good, thanks!

**Pull request merged**